internal/charmstore/resources_test.go

```diff
@@ -170,11 +170,16 @@ func (s *resourceSuite) TestListResourcesResourceNotFound(c *gc.C) {
 	sortResources(expected)
 	err := store.PublishResources(entity, channel, resourceRevisions(expected[1:]))
```

Comment:
I don't think PublishResources should allow publishing without all the resources
that are needed by the charm, so I'd write this test a bit differently.